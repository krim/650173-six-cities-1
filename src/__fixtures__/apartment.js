export default {
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 51.225402,
      longitude: 6.776314,
      zoom: 13
    }
  },
  previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
  images: [
    `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
  ],
  title: `Waterfront with extraordinary view`,
  isFavorite: false,
  isPremium: true,
  rating: 2.7,
  type: `hotel`,
  bedrooms: 2,
  maxAdults: 8,
  price: 393,
  goods: [
    `Breakfast`,
    `Washer`,
    `Laptop friendly workspace`
  ],
  host: {
    id: 25,
    name: `Angelina`,
    isPro: true,
    avatarUrl: `img/avatar-angelina.jpg`
  },
  description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
  location: {
    latitude: 51.214402,
    longitude: 6.764314000000001,
    zoom: 16
  },
  id: 1
};
