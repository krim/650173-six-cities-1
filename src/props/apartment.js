import PropTypes from 'prop-types';
import {range} from 'react-range-proptypes';

export default PropTypes.shape({
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  previewImage: PropTypes.string.isRequired,
  rating: range(0, 100).isRequired,
  price: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  bedrooms: PropTypes.number.isRequired,
  maxAdults: PropTypes.number.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  isPremium: PropTypes.bool.isRequired,
  location: PropTypes.shape({
    latitude: PropTypes.number.isRequired,
    longitude: PropTypes.number.isRequired,
    zoom: PropTypes.number.isRequired
  }),
  host: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    isPro: PropTypes.bool.isRequired,
    avatarUrl: PropTypes.string.isRequired
  })
});
