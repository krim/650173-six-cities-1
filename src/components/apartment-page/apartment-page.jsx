import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import getRating from '../../libs/get-rating';
import {userProps, apartmentProps, mapSettingsProps} from '../../props';
import {getNearApartments} from '../../reducer/data/selectors';
import ReviewList from '../review-list/review-list.jsx';
import ApartmentList from '../apartment-list/apartment-list.jsx';
import {ApartmentTypes, MaxCount} from '../../constants';
import BookmarkButton from '../bookmark-button/bookmark-button.jsx';
import Map from '../map/map.jsx';
import withFavorite from '../../hocs/with-favorite/with-favorite';
import withApartment from '../../hocs/with-apartment/with-apartment';
import Header from '../header/header.jsx';

const ApartmentPage = (props) => {
  const {
    user,
    apartment,
    nearApartments,
    mapSettings,
    onBookmarkClick,
    isUserAuthorized
  } = props;

  const {host} = apartment;

  return (
    <div className="page">
      <Header user={user} />
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {
                apartment.images.slice(0, MaxCount.APARTMENT_IMAGES).map((image, index) => {
                  return (
                    <div key={`image-${index}`} className="property__image-wrapper">
                      <img className="property__image" src={image} alt="Photo studio" />
                    </div>
                  );
                })
              }
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              { apartment.isPremium && <div className="property__mark"><span>Premium</span></div> }
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {apartment.title}
                </h1>
                <BookmarkButton
                  className={`property`}
                  height={`33`}
                  width={`31`}
                  isFavorite={apartment.isFavorite}
                  onBookmarkClick={onBookmarkClick}
                />
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: getRating(apartment.rating)}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">{apartment.rating}</span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {ApartmentTypes[apartment.type]}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {apartment.bedrooms} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                  Max {apartment.maxAdults} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{apartment.price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {
                    apartment.goods.map((advantage, index) => {
                      return (
                        <li key={`advantage-${index}`} className="property__inside-item">
                          {advantage}
                        </li>
                      );
                    })
                  }
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                    <img
                      className="property__avatar user__avatar"
                      src={host.avatarUrl}
                      width="74"
                      height="74"
                      alt="Host avatar"
                    />
                  </div>
                  <span className="property__user-name">
                    { host.name }
                  </span>
                  { host.isPro && <span className="property__user-status">Pro</span> }
                </div>
                <div className="property__description">
                  <p className="property__text">
                    { apartment.description }
                  </p>
                </div>
              </div>
              <ReviewList apartmentId={apartment.id} isUserAuthorized={isUserAuthorized} />
            </div>
          </div>
          {
            nearApartments.length > 0 &&
              <Map
                apartments={nearApartments.concat(apartment)}
                activeApartment={apartment}
                mapSettings={
                  {...mapSettings, location: apartment.city.location}
                }
                pageType={`property`}
              />
          }
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>
            <div className="near-places__list places__list">
              <ApartmentList
                apartments={nearApartments}
                setApartment={() => {}}
                onImageClick={() => {}}
                className={`cities`}
              />
            </div>
          </section>
        </div>
      </main>
    </div>
  );
};

ApartmentPage.propTypes = {
  user: userProps,
  mapSettings: mapSettingsProps,
  apartment: apartmentProps,
  nearApartments: PropTypes.arrayOf(apartmentProps),
  onBookmarkClick: PropTypes.func,
  isUserAuthorized: PropTypes.bool.isRequired
};

const mapStateToProps = (state, props) => {
  return {
    nearApartments: getNearApartments(state, props.match.params)
  };
};

const ApartmentPageWithState = connect(mapStateToProps, null)(ApartmentPage);

export {ApartmentPage, ApartmentPageWithState};
export default withApartment(
    withFavorite(ApartmentPageWithState)
);
