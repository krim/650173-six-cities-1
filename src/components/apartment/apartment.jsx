import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

import {apartmentProps} from '../../props';
import getRating from '../../libs/get-rating';

class Apartment extends PureComponent {
  render() {
    const {
      apartment,
      onClick,
      onMouseOver,
      onMouseOut
    } = this.props;

    return (
      <article
        onMouseOver={onMouseOver}
        onMouseOut={onMouseOut}
        className="cities__place-card place-card"
      >
        { apartment.isPremium && <div className="place-card__mark"><span>Premium</span></div> }
        <div className="cities__image-wrapper place-card__image-wrapper">
          <a href="#">
            <img
              className="place-card__image" src={apartment.previewImage} width="260" height="200" alt="Place image"
            />
          </a>
        </div>
        <div className="place-card__info">
          <div className="place-card__price-wrapper">
            <div className="place-card__price">
              <b className="place-card__price-value">&euro;{apartment.price}</b>
              <span className="place-card__price-text">&#47;&nbsp;night</span>
            </div>
            <button className="place-card__bookmark-button button" type="button">
              <svg className="place-card__bookmark-icon" width="18" height="19">
                <use xlinkHref="#icon-bookmark"></use>
              </svg>
              <span className="visually-hidden">To bookmarks</span>
            </button>
          </div>
          <div className="place-card__rating rating">
            <div className="place-card__stars rating__stars">
              <span style={{width: getRating(apartment.rating)}}></span>
              <span className="visually-hidden">Rating</span>
            </div>
          </div>
          <h2 className="place-card__name">
            <a href="#" onClick={onClick}>{ apartment.title }</a>
          </h2>
          <p className="place-card__type">{apartment.type}</p>
        </div>
      </article>
    );
  }
}

Apartment.propTypes = {
  apartment: apartmentProps,
  onClick: PropTypes.func.isRequired,
  onMouseOver: PropTypes.func.isRequired,
  onMouseOut: PropTypes.func.isRequired
};

export default Apartment;

