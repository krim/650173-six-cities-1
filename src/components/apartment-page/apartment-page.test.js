import React from 'react';
import renderer from 'react-test-renderer';

import {ApartmentPage} from './apartment-page';
import apartment from '../../__fixtures__/apartment';

const apartments = [apartment];

describe(`ApartmentPage`, () => {
  it(`renders component correctly`, () => {
    const tree = renderer.create(
        <ApartmentPage
          match={{params: {id: apartment.id.toString()}}}
          apartment={apartment}
          nearApartments={apartments}
          setApartmentId={jest.fn()}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
