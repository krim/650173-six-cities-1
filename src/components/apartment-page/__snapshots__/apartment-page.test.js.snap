// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApartmentPage renders component correctly 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://es31-server.appspot.com/six-cities/static/hotel/18.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://es31-server.appspot.com/six-cities/static/hotel/4.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://es31-server.appspot.com/six-cities/static/hotel/12.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://es31-server.appspot.com/six-cities/static/hotel/20.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://es31-server.appspot.com/six-cities/static/hotel/17.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://es31-server.appspot.com/six-cities/static/hotel/15.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Waterfront with extraordinary view
          </h1>
          <button
            className="property__bookmark-button button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            2.7
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Entire place
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            2
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            8
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            393
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Breakfast
            </li>
            <li
              className="property__inside-item"
            >
              Washer
            </li>
            <li
              className="property__inside-item"
            >
              Laptop friendly workspace
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
            <span
              className="property__user-status"
            >
              Pro
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews · 
            <span
              className="reviews__amount"
            >
              1
            </span>
          </h2>
          <ul
            className="reviews__list"
          >
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="https://es31-server.appspot.com/six-cities/static/avatar/4.jpg"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  Zak
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-05-21"
                >
                  May 2019
                </time>
              </div>
            </li>
          </ul>
          <form
            action="#"
            className="reviews__form form"
            method="post"
          >
            <label
              className="reviews__label form__label"
              htmlFor="review"
            >
              Your review
            </label>
            <div
              className="reviews__rating-form form__rating"
            >
              <input
                className="form__rating-input visually-hidden"
                id="5-stars"
                name="rating"
                type="radio"
                value="5"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="5-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="4-stars"
                name="rating"
                type="radio"
                value="4"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="4-stars"
                title="good"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="3-stars"
                name="rating"
                type="radio"
                value="3"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="3-stars"
                title="not bad"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="2-stars"
                name="rating"
                type="radio"
                value="2"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="2-stars"
                title="badly"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="1-star"
                name="rating"
                type="radio"
                value="1"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="1-star"
                title="terribly"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
            </div>
            <textarea
              className="reviews__textarea form__textarea"
              id="review"
              name="review"
              placeholder="Tell how was your stay, what you like and what can be improved"
            />
            <div
              className="reviews__button-wrapper"
            >
              <p
                className="reviews__help"
              >
                To submit review please make sure to set 
                <span
                  className="reviews__star"
                >
                  rating
                </span>
                 and describe your stay with at least 
                <b
                  className="reviews__text-amount"
                >
                  50 characters
                </b>
                .
              </p>
              <button
                className="reviews__submit form__submit button"
                disabled=""
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
    <section
      className="property__map map"
      id="map"
    />
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      >
        <article
          className="cities__place-card place-card"
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://es31-server.appspot.com/six-cities/static/hotel/1.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  393
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "60%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                Waterfront with extraordinary view
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Hotel
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;
